set_1_basics/challenge_1_convert_hex_to_base64/encode_base64.cpp

```diff
@@ -29,51 +29,43 @@ std::ostream & operator<<(std::ostream & output,
     }
 
     // First base64 digit is composed of high 6 bits of first octet
-    output <<
-            encode_base64::convertToBase64(
-                    *manipulator.d_input[0]
-                    >> (
-                            encode_base64::s_numBitsInOctet
-                            - encode_base64::s_numBitsInBase64Digit
-                       )
-            );
+    output << encode_base64::convertToBase64(
+        *manipulator.d_input[0] >> (encode_base64::s_numBitsInOctet
+                                    - encode_base64::s_numBitsInBase64Digit));
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Do not access "s_numBitsInBase64Digit" with derived type "encode_base64". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS3252)